src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -510,6 +510,43 @@ private void Get_CORINFO_SIG_INFO(MethodDesc method, CORINFO_SIG_INFO* sig, bool
             }
         }
 
+        private bool TryGetUnmanagedCallingConventionFromModOpt(MethodSignature signature, out CorInfoCallConv callConv)
+        {
+            callConv = CorInfoCallConv.CORINFO_CALLCONV_UNMANAGED;
+            if (!signature.HasEmbeddedSignatureData || signature.GetEmbeddedSignatureData() == null)
+                return false;
+
+            foreach (EmbeddedSignatureData data in signature.GetEmbeddedSignatureData())
```

Comment:
This should be looking at the custom modifier on the return parameter only.

If someone places a modifier on one of the parameters, it should be ignored. I think this can actually be hit with C# - a method signature that has a function pointer as one of the parameters would have multiple modifiers in it.

I don't know how to extract that information from `EmbeddedSignatureData`. It's a ridiculously bad API and it should have never been merged into the type system. I get unreasonably angry whenever I see it. @davidwrighton might have advice.

Reply:
Boo, totally missed that. Could the `EmbeddedSignatureData.index` string be an indicator? From looking through how it is set, it looks like the modifiers for the return type should have the 'lowest' index, but I'm not entirely sure what exactly that is representing / how to parse that?

@davidwrighton - would appreciate any guidance. I'd like to get this change done before the P8 snap tomorrow.

Reply:
Hmm... yes, the api is the worst. (but it does ... work... blech.) It represents the hierarchy of parsing the signature, and it actually allows for substitution inside of the signature without breaking stuff (which I must say is somewhat miraculous).

However, the goal was to require the modopt for these to be at the very beginning of the modifier stream before any non mod opts. Fortunately, that will result in the index following a consistent pattern. I need to do some research to understand the actual pattern you need to actually work with. I hope I can get back to you today.

Reply:
Ok, so, you'll want to only examine those modopts which have exactly the index of `0.1.1.1` I've just filed #39329 which adds a test to ensure that this behavior remains consistent.

Reply:
You might also consider adding an constant string to the MethodSignature type named `IndexOfCustomModifiersOnReturnType` and put the magic string in there. I don't like the idea of blindly writing the literal string `0.1.1.1` here.